dali/operators/decoder/audio/audio_decoder_op.cc

```diff
@@ -13,11 +13,14 @@
 // limitations under the License.
 
 #include "dali/operators/decoder/audio/audio_decoder_op.h"
+#include "dali/operators/decoder/audio/audio_decoder_impl.h"
 #include "dali/pipeline/operator/op_schema.h"
 #include "dali/pipeline/data/views.h"
 
 namespace dali {
 
+using DecoderType = int16_t;
```

Comment:
This is not a good name. `DecodedType` or `DecoderOutputType` would explain it much better.

Reply:
I know that for ASR 16 bit or even 8-bit would probably do - but not so much for other signals. 24-bit audio is becoming common and 32-bit is not unheard of.

Reply:
I see that you've removed the TYPE_SWITCH that enabled decoding to the requested output type.
NO GO.

Reply:
Restored the type-switch